File: Stack/ArrayImplementation/stack.js
class Stack{
    constructor(){
        this.items=[]
        this.t=-1
        this.MAX=5
    }

    isFull(){
        if(this.t == this.MAX)
          return true
        return false  
    }

    isEmpty(){
        if(this.t < 0)
          return true
        return false  
    }

    push(data){
        //Check if array is Full
        if(this.isFull()){
            console.log("Array is Full")
            return
        }
        this.t=this.t+1
        this.items.push(data)
    }

    pop(){
       //check if array is Empty
       if(this.isEmpty()){
           console.log("Array Is Empty")
           return
       }
       let val=this.items.pop()
       console.log(`Element Removed is ${val}`)
       this.t=this.t-1
    }

    printStack(){
        console.log("Will Print Stack From Top")
        let i=this.t
        while(i>=0){
            console.log(this.items[i])
            i--
        }
        console.log("\n")
    }
}

let stack= new Stack()
stack.push(10)
stack.push(20)
stack.push(30)
stack.printStack()
stack.pop()
stack.printStack()

File: Stack/ArrayImplementation/stack.py
class Stack:
    def __init__(self):
        self.items=[]
        self.top=-1
        self.MAX=5

    def isFull(self):
        if(self.top == self.MAX):
            return True    
        return False 
    def isEmpty(self):
        if(self.top < -1):
            return True
        return False

    def Push(self,data):
        if(self.isFull()):
            print("Stack Is Full")
            return
        self.top=self.top+1
        self.items.append(data)

    def Pop(self):
        if(self.isEmpty()):
            print("Stack Is Empty")
            return
        val=self.items.pop()
        print("Item Removed is ",val)
        self.top=self.top-1

    def printStack(self):
        for i in range(0,self.top+1):
            print(self.items[i])  

st=Stack()

st.Push(10)
st.Push(20)
st.Push(30)
st.printStack()

st.Pop()
st.printStack()